Add offline maps

pronebird opened the pull request:
This PR adds offline SVG maps

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/mullvad/mullvadvpn-app/45)
<!-- Reviewable:end -->

eriklarko commented:
Reviewed 11 of 46 files at r1.
Review status: 11 of 46 files reviewed at latest revision, 1 unresolved discussion.

---

*[app/components/Connect.js, line 138 at r1](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3hdcodJd5SEf4sp_C9:-L3hdcodJd5SEf4sp_CA:bn8iqr4) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/ce13641c426feb8cdb6677ce99ed6717f5fa29e3/app/components/Connect.js#L138)):*
> ```JavaScript
>     const { longitude, latitude } = this.props.connection;
> 
>     let zoomLevel, center, offset, showUserLocation;
> ```

Perhaps this could to into a separate method?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 10 of 46 files reviewed at latest revision, 1 unresolved discussion.

---

*[app/components/Connect.js, line 138 at r1](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3hdcodJd5SEf4sp_C9:-L3hi0RXOOULMT0Y2zcm:b-896fix) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/ce13641c426feb8cdb6677ce99ed6717f5fa29e3/app/components/Connect.js#L138)):*
<details><summary><i>Previously, eriklarko (Erik Larkö) wrote…</i></summary><blockquote>

Perhaps this could to into a separate method?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

anderklander commented:
Review status: 10 of 25 files reviewed at latest revision, 3 unresolved discussions.

---

*[package.json, line 24 at r2](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3hpBeLvfmhD2_MH38k:-L3hpBeLvfmhD2_MH38l:b-h8krtx) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/4c6d971865662d6636c7ce7536fcbf4350696708/package.json#L24)):*
> ```JSON
>     "react": "^16.0.0",
>     "react-dom": "^16.0.0",
>     "react-native": "^0.50.4",
> ```

We will need react-native >53 when rewriting the AccountInput.

---

*[app/components/Connect.js, line 152 at r2](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3hpiX7-fAiq0HgdLHd:-L3hpiX7-fAiq0HgdLHe:b-djptrc) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/4c6d971865662d6636c7ce7536fcbf4350696708/app/components/Connect.js#L152)):*
> ```JavaScript
> 
>         // a magic offset to align marker with spinner
>         offset: [0, 123],
> ```

Is this some pixel-offset? Just trying to think ahead if running on different screen sizes on mobile. (and not sure if we can use the same maps at all ... so dont spend to much thought)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 10 of 25 files reviewed at latest revision, 3 unresolved discussions.

---

*[package.json, line 24 at r2](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3hpBeLvfmhD2_MH38k:-L3hqrVE0peAgcV5JLT-:b-djffk4) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/4c6d971865662d6636c7ce7536fcbf4350696708/package.json#L24)):*
<details><summary><i>Previously, anderklander wrote…</i></summary><blockquote>

We will need react-native >53 when rewriting the AccountInput.
</blockquote></details>

Do you want me to update it in this PR?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 10 of 25 files reviewed at latest revision, 3 unresolved discussions.

---

*[app/components/Connect.js, line 152 at r2](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3hpiX7-fAiq0HgdLHd:-L3hr-is4FsRhzTPj3b4:b-nxln8h) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/4c6d971865662d6636c7ce7536fcbf4350696708/app/components/Connect.js#L152)):*
<details><summary><i>Previously, anderklander wrote…</i></summary><blockquote>

Is this some pixel-offset? Just trying to think ahead if running on different screen sizes on mobile. (and not sure if we can use the same maps at all ... so dont spend to much thought)
</blockquote></details>

Yes it is. This map is for desktop only. It's highly unlikely we'll be able to reuse it for mobile. Also mobile offers native maps which are way faster but not sure about how customizable they are.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

anderklander commented:
Review status: 10 of 25 files reviewed at latest revision, 3 unresolved discussions.

---

*[package.json, line 24 at r2](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3hpBeLvfmhD2_MH38k:-L3hsAxEFgKeV4V33iLA:b-xr7adp) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/4c6d971865662d6636c7ce7536fcbf4350696708/package.json#L24)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

Do you want me to update it in this PR?
</blockquote></details>

I think that its best if a PR regarding maps for desktop don't change the react-native version without a reason, so yes please!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

anderklander commented:
Review status: 8 of 25 files reviewed at latest revision, 3 unresolved discussions.

---

*[app/components/Connect.js, line 152 at r2](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3hpiX7-fAiq0HgdLHd:-L3hwEUN5WdkDkjCeAVC:b-cusmyq) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/4c6d971865662d6636c7ce7536fcbf4350696708/app/components/Connect.js#L152)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

Yes it is. This map is for desktop only. It's highly unlikely we'll be able to reuse it for mobile. Also mobile offers native maps which are way faster but not sure about how customizable they are.
</blockquote></details>

Could this be moved to Map.js? I guess also the zoomLevel is dependent on map/framework and would be better if integrated in the map component?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 8 of 25 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[package.json, line 24 at r2](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3hpBeLvfmhD2_MH38k:-L3i-zbpEcULF7D9uWnL:b-64xt9y) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/4c6d971865662d6636c7ce7536fcbf4350696708/package.json#L24)):*
<details><summary><i>Previously, anderklander wrote…</i></summary><blockquote>

I think that its best if a PR regarding maps for desktop don't change the react-native version without a reason, so yes please!
</blockquote></details>

Oh wait I think I messed up during the rebase. Sorry for that.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 8 of 25 files reviewed at latest revision, 3 unresolved discussions.

---

*[app/components/Connect.js, line 152 at r2](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3hpiX7-fAiq0HgdLHd:-L3i15lTB4hUQGIsrjGs:b-r5sibc) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/4c6d971865662d6636c7ce7536fcbf4350696708/app/components/Connect.js#L152)):*
<details><summary><i>Previously, anderklander wrote…</i></summary><blockquote>

Could this be moved to Map.js? I guess also the zoomLevel is dependent on map/framework and would be better if integrated in the map component?
</blockquote></details>

We could probably introduce three zoom level constants and move the actual values into the Map

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 8 of 25 files reviewed at latest revision, 3 unresolved discussions.

---

*[package.json, line 24 at r2](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3hpBeLvfmhD2_MH38k:-L3i1GCT4wRkpE8XowNY:bpul4jp) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/4c6d971865662d6636c7ce7536fcbf4350696708/package.json#L24)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

Oh wait I think I messed up during the rebase. Sorry for that.
</blockquote></details>

That should be fixed now for all react-native-* deps. Clearly was not intentional.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 8 of 25 files reviewed at latest revision, 3 unresolved discussions.

---

*[app/components/Map.js, line 304 at r3](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3i1qDH6PadyD_KWCRf:-L3i1qDH6PadyD_KWCRg:b-j2upm) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/b4184662a3140ed289d132f4ad8b0f47ed60c56b/app/components/Map.js#L304)):*
> ```JavaScript
> 
>     // combine previous and current viewports to get the rough area of transition
>     const combinedViewportBboxMatch = prevState ? {
> ```

The viewport is a visible part of the map and is a rectangle. Imagine this rectangle moves across the map from point A to point B.

Now if we would draw the lines connecting each of the vertices of the each of the rectangle, we could find the trajectory of "camera".

We could reduce the set first to union bbox of source and destination points and then do the second pass to filter out the geometry that does not intersect with the trajectory lines. This would dramatically reduce the amount of drawn geometry on diagonal travels.

This is something that came to my mind as a future optimization but there may be better ways to achieve the same... :)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 8 of 25 files reviewed at latest revision, 4 unresolved discussions.

---

*[app/components/Map.js, line 316 at r3](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3i4Zzp6QnelJeG156E:-L3i4Zzp6QnelJeG156F:barbop4) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/b4184662a3140ed289d132f4ad8b0f47ed60c56b/app/components/Map.js#L316)):*
> ```JavaScript
>     };
> 
>     const visibleCountries = zoomLevel < 5 || zoomLevel > 20 ? [] : countryTree.search(viewportBboxMatch);
> ```

To reduce the number of rendered SVG nodes I query the R-tree and find the visible shapes. This is later used in render()

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 8 of 25 files reviewed at latest revision, 5 unresolved discussions.

---

*[app/components/Map.js, line 136 at r3](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3i4pxRbjX7zSUz5gC-:-L3i4pxRbjX7zSUz5gC0:b-glggoh) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/b4184662a3140ed289d132f4ad8b0f47ed60c56b/app/components/Map.js#L136)):*
> ```JavaScript
>         marker={{ coordinates: this.props.center }}
>         style={ markerStyle }>
>         <image x="-30" y="-30" href={ this.props.markerImagePath } />
> ```

the marker itself is 60x60 so we need to shift the anchor point to draw it in the middle.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 8 of 25 files reviewed at latest revision, 6 unresolved discussions.

---

*[app/components/Map.js, line 173 at r3](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3i4y-VttEkiHU5_W6C:-L3i4y-VttEkiHU5_W6D:b-tg3ri1) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/b4184662a3140ed289d132f4ad8b0f47ed60c56b/app/components/Map.js#L173)):*
> ```JavaScript
>           disablePanning={ false }
>           style={ zoomableGroupStyle }>
>           <Geographies geography={ geographyData } disableOptimization={ true }>
> ```

The first layer of geometries renders continents and countries

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 8 of 25 files reviewed at latest revision, 7 unresolved discussions.

---

*[app/components/Map.js, line 188 at r3](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3i5-vA3xPyWDi1TMKK:-L3i5-vA3xPyWDi1TMKL:beb0m1i) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/b4184662a3140ed289d132f4ad8b0f47ed60c56b/app/components/Map.js#L188)):*
> ```JavaScript
>             }}
>           </Geographies>
>           <Geographies geography={ statesProvincesLinesData } disableOptimization={ true }>
> ```

The second layer renders state and provinces lines (borders) 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 8 of 25 files reviewed at latest revision, 8 unresolved discussions.

---

*[app/components/Map.js, line 246 at r3](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3i5AiASJSVlmJOiuTl:-L3i5AiASJSVlmJOiuTm:b1a0z08) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/b4184662a3140ed289d132f4ad8b0f47ed60c56b/app/components/Map.js#L246)):*
> ```JavaScript
>     zoom: number
>   ) {
>     const pos = projection(center);
> ```

The actual center of the map is a little off to make sure that we render the marker a little higher to align with spinner.

This method does the right calculates to compute the actual geo center of the map that we pass down to react-simple-maps.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 8 of 25 files reviewed at latest revision, 9 unresolved discussions.

---

*[app/components/Map.js, line 253 at r3](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3i5djYnRmbXKId0jTJ:-L3i5djYnRmbXKId0jTK:b-u7orrr) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/b4184662a3140ed289d132f4ad8b0f47ed60c56b/app/components/Map.js#L253)):*
> ```JavaScript
>   }
> 
>   _getViewportGeoBoundingBox(
> ```

This method calculates the visible geo boundaries.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 8 of 25 files reviewed at latest revision, 10 unresolved discussions.

---

*[geo-data/extract-countries.py, line 28 at r3](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3i79gFXW2aYuEVz8ad:-L3i79gFXW2aYuEVz8ae:b-tmllz5) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/b4184662a3140ed289d132f4ad8b0f47ed60c56b/geo-data/extract-countries.py#L28)):*
> ```Python
> 
>     # lowercase all keys
>     props = dict((k.lower(), v) for k, v in feat["properties"].iteritems())
> ```

Some shape data comes with uppercase property keys some not.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 8 of 25 files reviewed at latest revision, 11 unresolved discussions.

---

*[geo-data/extract-cities.py, line 20 at r3](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3i7Fw7gHo3BNQO_yrg:-L3i7Fw7gHo3BNQO_yrh:b-5pv206) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/b4184662a3140ed289d132f4ad8b0f47ed60c56b/geo-data/extract-cities.py#L20)):*
> ```Python
>   for feat in source:
>     props = feat["properties"]
>     if props["scalerank"] < 8:
> ```

I filter out the small cities. I don't actually remember what the highest scale-rank is but 8 seemed to be like a good match after testing.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

eriklarko commented:
Reviewed 10 of 46 files at r1, 1 of 23 files at r2, 4 of 4 files at r3, 1 of 1 files at r4.
Review status: 23 of 25 files reviewed at latest revision, 15 unresolved discussions.

---

*[app/components/Connect.js, line 138 at r1](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3hdcodJd5SEf4sp_C9:-L3lKtbAKvveYjN3ZWRa:b-83sjoh) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/ce13641c426feb8cdb6677ce99ed6717f5fa29e3/app/components/Connect.js#L138)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Nice!

---

*[geo-data/extract-cities.py, line 1 at r4](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3lbISeGFkgNqpVksYP:-L3lbISfp4N8IwBggYEN:bgv3sn8) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/6cd193f3c11f93b972324c61169489914bb05b1a/geo-data/extract-cities.py#L1)):*
> ```Python
> """
> ```

Will we want to be able to update just one part of our geometries or would it perhaps be easier if we merged these four modules into one thing that does them all? 

---

*[geo-data/README, line 1 at r4](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3lahErFyRqXYW2Dnza:-L3lahEs7iU2waTAyuQ7:b1nlij9) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/6cd193f3c11f93b972324c61169489914bb05b1a/geo-data/README#L1)):*
> ```
> This is a folder with python 2 and node scripts to produce geographical
> ```

Nice readme!

---

*[geo-data/README, line 64 at r4](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3l_qlvxnx7xxhJJwrh:-L3l_qlwOD8Q-gg8jo3l:b-3q1ig3) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/6cd193f3c11f93b972324c61169489914bb05b1a/geo-data/README#L64)):*
> ```
> 
> ```
> yarn babel-node geo-data/prepare-rtree.js
> ```

We use npm above, and though I very much prefer yarn it might be easier to follow these instructions later if they use the same tool

---

*[geo-data/README, line 76 at r4](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3la6rVMLf9lCZQJ-LN:-L3la6rVMLf9lCZQJ-LO:bszv5kw) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/6cd193f3c11f93b972324c61169489914bb05b1a/geo-data/README#L76)):*
> ```
> - geometry.rbush.json
> - states-provinces-lines.json
> - states-provinces-lines.rbush.json
> ```

It could be nice to put all this into a bash script so that we can run e.g. `./update-geometries.sh`. But we won't do it very often so it might not be worth it :) Just a thought

---

*[geo-data/requirements.txt, line 1 at r4](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3lalku7lWbV3-9jLr9:-L3lalku7lWbV3-9jLrA:b-xgezld) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/6cd193f3c11f93b972324c61169489914bb05b1a/geo-data/requirements.txt#L1)):*
> ```
> fiona==1.7.11
> ```

This might be paranoid, but it would feel safer if we add the hash so that we can be sure that we get the exact same dependency every time. https://pip.pypa.io/en/stable/user_guide/#hash-checking-mode

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

eriklarko commented:
Reviewed 1 of 46 files at r1.
Review status: 24 of 25 files reviewed at latest revision, 13 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 24 of 25 files reviewed at latest revision, 13 unresolved discussions.

---

*[geo-data/README, line 64 at r4](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45#-L3l_qlvxnx7xxhJJwrh:-L3mGXH77_VTts78u9LR:b30jz7j) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/6cd193f3c11f93b972324c61169489914bb05b1a/geo-data/README#L64)):*
<details><summary><i>Previously, eriklarko (Erik Larkö) wrote…</i></summary><blockquote>

We use npm above, and though I very much prefer yarn it might be easier to follow these instructions later if they use the same tool
</blockquote></details>

`npx babel-node prepare-tree.js` should do too.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 24 of 25 files reviewed at latest revision, 13 unresolved discussions.

---

*[geo-data/README, line 64 at r4](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45#-L3l_qlvxnx7xxhJJwrh:-L3mIEJf6A-c5JACb1hd:b-896fix) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/6cd193f3c11f93b972324c61169489914bb05b1a/geo-data/README#L64)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

`npx babel-node prepare-tree.js` should do too.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

anderklander commented:
Review status: 23 of 25 files reviewed at latest revision, 12 unresolved discussions.

---

*[app/components/Connect.js, line 152 at r2](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3hpiX7-fAiq0HgdLHd:-L3mTgD8Eg92d59kM-EG:b-107llv) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/4c6d971865662d6636c7ce7536fcbf4350696708/app/components/Connect.js#L152)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

We could probably introduce three zoom level constants and move the actual values into the Map
</blockquote></details>

I think it would be very good if all implementation-specific constants were moved to the map component. Just zoomLevel and offset?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 18 of 23 files reviewed at latest revision, 12 unresolved discussions.

---

*[geo-data/README, line 76 at r4](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45#-L3la6rVMLf9lCZQJ-LN:-L3mJdug0bxBjDsQqxen:b-byf4vk) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/6cd193f3c11f93b972324c61169489914bb05b1a/geo-data/README#L76)):*
<details><summary><i>Previously, eriklarko (Erik Larkö) wrote…</i></summary><blockquote>

It could be nice to put all this into a bash script so that we can run e.g. `./update-geometries.sh`. But we won't do it very often so it might not be worth it :) Just a thought
</blockquote></details>

I've added an integration python script.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

eriklarko commented:
*[geo-data/README, line 76 at r4](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3la6rVMLf9lCZQJ-LN:-L3maVk95AO2CpQ85ZmS:bzftu3a) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/6cd193f3c11f93b972324c61169489914bb05b1a/geo-data/README#L76)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

I've added an integration python script.
</blockquote></details>

Sweet!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 18 of 23 files reviewed at latest revision, 12 unresolved discussions.

---

*[geo-data/requirements.txt, line 1 at r4](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45#-L3lalku7lWbV3-9jLr9:-L3marua4Hb_O55Fh8AP:bpfagtq) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/6cd193f3c11f93b972324c61169489914bb05b1a/geo-data/requirements.txt#L1)):*
<details><summary><i>Previously, eriklarko (Erik Larkö) wrote…</i></summary><blockquote>

This might be paranoid, but it would feel safer if we add the hash so that we can be sure that we get the exact same dependency every time. https://pip.pypa.io/en/stable/user_guide/#hash-checking-mode
</blockquote></details>

It doesn't work.

```
pip install --require-hashes -r requirements.txt
```

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 18 of 23 files reviewed at latest revision, 12 unresolved discussions.

---

*[geo-data/requirements.txt, line 1 at r4](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45#-L3lalku7lWbV3-9jLr9:-L3mbmk_DtyL4YExKJeY:buetdf0) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/6cd193f3c11f93b972324c61169489914bb05b1a/geo-data/requirements.txt#L1)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

It doesn't work.

```
pip install --require-hashes -r requirements.txt
```
</blockquote></details>

pip 9.0.1 over here

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 18 of 23 files reviewed at latest revision, 11 unresolved discussions.

---

*[geo-data/extract-cities.py, line 1 at r4](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45#-L3lbISeGFkgNqpVksYP:-L3mc23f4YbGFvWI9XfN:blu689n) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/6cd193f3c11f93b972324c61169489914bb05b1a/geo-data/extract-cities.py#L1)):*
<details><summary><i>Previously, eriklarko (Erik Larkö) wrote…</i></summary><blockquote>

Will we want to be able to update just one part of our geometries or would it perhaps be easier if we merged these four modules into one thing that does them all? 
</blockquote></details>

It depends, but I merged all of them into one.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

eriklarko commented:
Reviewed 6 of 8 files at r6.
Review status: 22 of 23 files reviewed at latest revision, 12 unresolved discussions.

---

*[geo-data/extract-geo-data.py, line 144 at r6](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3mc9lrcNUmkziOJorI:-L3mc9lrcNUmkziOJorJ:b-9px4g7) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/dc5cf4bb072e3f81924bc84b7b8bd7e9f49ae841/geo-data/extract-geo-data.py#L144)):*
> ```Python
> extract_countries()
> extract_geometry()
> extract_provinces_and_states_lines()
> ```

Aaaaaw yiiiieah. Awesome

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

eriklarko commented:
*[geo-data/requirements.txt, line 1 at r4](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3lalku7lWbV3-9jLr9:-L3mcN7sl7z4zvEW-ngP:bz5wz8v) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/6cd193f3c11f93b972324c61169489914bb05b1a/geo-data/requirements.txt#L1)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

pip 9.0.1 over here
</blockquote></details>

Hmmm.. What kind of error are you getting? Perhaps https://davidwalsh.name/hashin could help? If not then perhaps we can ignore this. We've got bigger fish :)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

eriklarko commented:
Reviewed 1 of 46 files at r1.
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[geo-data/requirements.txt, line 1 at r4](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45#-L3lalku7lWbV3-9jLr9:-L3meBBbCuMi0K2SfSxP:bpoebjf) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/6cd193f3c11f93b972324c61169489914bb05b1a/geo-data/requirements.txt#L1)):*
<details><summary><i>Previously, eriklarko (Erik Larkö) wrote…</i></summary><blockquote>

Hmmm.. What kind of error are you getting? Perhaps https://davidwalsh.name/hashin could help? If not then perhaps we can ignore this. We've got bigger fish :)
</blockquote></details>

It just says "Requirement already satisfied" and doesn't output the hashes like they feature in the documentation.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[geo-data/requirements.txt, line 1 at r4](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45#-L3lalku7lWbV3-9jLr9:-L3memR5BPdipPxLZVPd:b-a16mff) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/6cd193f3c11f93b972324c61169489914bb05b1a/geo-data/requirements.txt#L1)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

It just says "Requirement already satisfied" and doesn't output the hashes like they feature in the documentation.
</blockquote></details>

Is it a hash per platform? Have any clue?

```
Fiona==1.7.11 \
    --hash=sha256:c9becd2b450599ed59bd21418325ea0a388690822765de6ce2813049a53cd50e \
    --hash=sha256:7e438a7f9e63ecac81074136cb821a76b8c9e160bd5f03fb4fea78651f78036f \
    --hash=sha256:c2d9afb8b6cdc80b7e6e77872041dc61fb96a938007a6ee51703893d93ab76a6 \
    --hash=sha256:51b48f04038303d4aaccda5d53e13c5b5cb0e7232b9a0f21d588be9ec964447b \
    --hash=sha256:9ce609746cd70dfe071db966d8c08788724eb8878b64a06db71775a5bb1e6937 \
    --hash=sha256:305b893dbf06962e77addc7157e217817e352b764c248d2557ea41745406d754 \
    --hash=sha256:1d8a9d35cf951008477c69ea9539a678f90d3b775dad9eb08023d9e3a9137afa \
    --hash=sha256:c6e236950444d8efe0a78a1c17f31941c7930ef72f6026e5e6a15ef9601e8c0b \
    --hash=sha256:0079db16bb6d826cb43f841bdce452c0d68c381a9ec0a44a21e78982bc421538 \
    --hash=sha256:5e9c68ea71e9d79fcfb68c9a101c0b133301e233c9bcca7b7c65f33cc7636ef5
Shapely==1.6.3 \
    --hash=sha256:db8e5e512824c58084092a644a6256b404f5e6f24b48f635ed86bc9c6e6299fa \
    --hash=sha256:32e2e45994c3d757fa09e40538aa59d1d070375599ac149ba30d80ce2dd2c5af \
    --hash=sha256:f0edcea8ae2f3cc17ea5f7bbe466250681498045b6e99bb570e619eb8bbc82f7 \
    --hash=sha256:13ce19e4b7ea872a1deec007e797bf16df9cc5b8df0e7b4c7b5ef8e1efc9209f \
    --hash=sha256:ed3938fdd088bfb2b536d80cea503abcb997f0aece3f13ccfcdb83e78b18e65e \
    --hash=sha256:3f25d057edef86ece556ef94e2473bfdc17ea32714a0b6c3f77435e8f94bb8b0 \
    --hash=sha256:d25fe725780029cf9300d637b4b54b1b19f3a752a8cb6d209fc756ea72ebfc86 \
    --hash=sha256:87d424d35bbc1e92860f9f31849966a4be418839c4ea8e5a26459f72b1f10f5f \
    --hash=sha256:6fb5f62c319381b42e067e5bd6cbb6ad5a69c1f8915158554a8bc9dca6a65f1a \
    --hash=sha256:14152f111c7711fc6756fd538ec12fc8cdde7419f869b244922f71f61b2a6c6b

```

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[app/components/Connect.js, line 152 at r2](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45#-L3hpiX7-fAiq0HgdLHd:-L3mf4cI5eIIhRzq6ofq:b3y0spm) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/4c6d971865662d6636c7ce7536fcbf4350696708/app/components/Connect.js#L152)):*
<details><summary><i>Previously, anderklander wrote…</i></summary><blockquote>

I think it would be very good if all implementation-specific constants were moved to the map component. Just zoomLevel and offset?
</blockquote></details>

So what you're saying is the `Map` should know where the `Connect` component wants to render the spinner and adjust its offset accordingly?

As of zoom levels, it really depends on how zoom levels are defined on other platforms and maps solutions. 

For example we could calculate the zoom level based on altitude. Using altitude we could then align all maps to behave the same. 

We can cross this bridge later since the first implementation for mobile will most likely have no maps in the beginning.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[geo-data/requirements.txt, line 1 at r4](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45#-L3lalku7lWbV3-9jLr9:-L3mhuP-83f51IKLh3_8:bjh7qbf) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/6cd193f3c11f93b972324c61169489914bb05b1a/geo-data/requirements.txt#L1)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

Is it a hash per platform? Have any clue?

```
Fiona==1.7.11 \
    --hash=sha256:c9becd2b450599ed59bd21418325ea0a388690822765de6ce2813049a53cd50e \
    --hash=sha256:7e438a7f9e63ecac81074136cb821a76b8c9e160bd5f03fb4fea78651f78036f \
    --hash=sha256:c2d9afb8b6cdc80b7e6e77872041dc61fb96a938007a6ee51703893d93ab76a6 \
    --hash=sha256:51b48f04038303d4aaccda5d53e13c5b5cb0e7232b9a0f21d588be9ec964447b \
    --hash=sha256:9ce609746cd70dfe071db966d8c08788724eb8878b64a06db71775a5bb1e6937 \
    --hash=sha256:305b893dbf06962e77addc7157e217817e352b764c248d2557ea41745406d754 \
    --hash=sha256:1d8a9d35cf951008477c69ea9539a678f90d3b775dad9eb08023d9e3a9137afa \
    --hash=sha256:c6e236950444d8efe0a78a1c17f31941c7930ef72f6026e5e6a15ef9601e8c0b \
    --hash=sha256:0079db16bb6d826cb43f841bdce452c0d68c381a9ec0a44a21e78982bc421538 \
    --hash=sha256:5e9c68ea71e9d79fcfb68c9a101c0b133301e233c9bcca7b7c65f33cc7636ef5
Shapely==1.6.3 \
    --hash=sha256:db8e5e512824c58084092a644a6256b404f5e6f24b48f635ed86bc9c6e6299fa \
    --hash=sha256:32e2e45994c3d757fa09e40538aa59d1d070375599ac149ba30d80ce2dd2c5af \
    --hash=sha256:f0edcea8ae2f3cc17ea5f7bbe466250681498045b6e99bb570e619eb8bbc82f7 \
    --hash=sha256:13ce19e4b7ea872a1deec007e797bf16df9cc5b8df0e7b4c7b5ef8e1efc9209f \
    --hash=sha256:ed3938fdd088bfb2b536d80cea503abcb997f0aece3f13ccfcdb83e78b18e65e \
    --hash=sha256:3f25d057edef86ece556ef94e2473bfdc17ea32714a0b6c3f77435e8f94bb8b0 \
    --hash=sha256:d25fe725780029cf9300d637b4b54b1b19f3a752a8cb6d209fc756ea72ebfc86 \
    --hash=sha256:87d424d35bbc1e92860f9f31849966a4be418839c4ea8e5a26459f72b1f10f5f \
    --hash=sha256:6fb5f62c319381b42e067e5bd6cbb6ad5a69c1f8915158554a8bc9dca6a65f1a \
    --hash=sha256:14152f111c7711fc6756fd538ec12fc8cdde7419f869b244922f71f61b2a6c6b

```
</blockquote></details>

OK yes! it's a hash per python per platform.

```
hashin fiona==1.7.11 --python-version 2.7 --verbose
hashin shapely==1.6.3 --python-version 2.7 --verbose
```

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 22 of 23 files reviewed at latest revision, 2 unresolved discussions.

---

*[geo-data/requirements.txt, line 1 at r4](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45#-L3lalku7lWbV3-9jLr9:-L3miHnOBN_RvuyQNzdh:bxoes0e) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/6cd193f3c11f93b972324c61169489914bb05b1a/geo-data/requirements.txt#L1)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

OK yes! it's a hash per python per platform.

```
hashin fiona==1.7.11 --python-version 2.7 --verbose
hashin shapely==1.6.3 --python-version 2.7 --verbose
```
</blockquote></details>

Done in a separate commit just to remind myself how to run that thing :D If you wish I can rebase it into the first commit that adds this file.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

eriklarko commented:
Reviewed 1 of 1 files at r7.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[geo-data/requirements.txt, line 1 at r4](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3lalku7lWbV3-9jLr9:-L3mmYNROfosdulLGksO:b-qn5lbb) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/6cd193f3c11f93b972324c61169489914bb05b1a/geo-data/requirements.txt#L1)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

Done in a separate commit just to remind myself how to run that thing :D If you wish I can rebase it into the first commit that adds this file.
</blockquote></details>

Perfect!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

anderklander commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[app/components/Connect.js, line 152 at r2](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3hpiX7-fAiq0HgdLHd:-L3mkg44AX7xXy3MLROs:behfq79) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/4c6d971865662d6636c7ce7536fcbf4350696708/app/components/Connect.js#L152)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

So what you're saying is the `Map` should know where the `Connect` component wants to render the spinner and adjust its offset accordingly?

As of zoom levels, it really depends on how zoom levels are defined on other platforms and maps solutions. 

For example we could calculate the zoom level based on altitude. Using altitude we could then align all maps to behave the same. 

We can cross this bridge later since the first implementation for mobile will most likely have no maps in the beginning.
</blockquote></details>

No, i am saying that if the connect component wants to render something in the map component it should say that in something more logical than a magic offset of [0, 123], and something that can be made platform specific in a fairly easy way, the same goes for height/width, there are functions for measuring in reactxp https://microsoft.github.io/reactxp/docs/apis/userinterface.html, or if the hard-coded numbers can be just inside map instead we can make mobile work without having to change the desktop-version.

If zoomLevel is just depending on status, and if there is lat/lng i dont see why we should have some number that does not have a meaning if using another map. Enum or just levels that reflect what the purpose is (or just send the props needed for the logic to the map)

If we just use something that is not looking completely random it would be great, if we dont feel good with just sending the status and let the map calculate i think it would be much better to use 

Yes, everything can be done later. But if we do everything later it will never become finished. The rest of the connect component would be great if we could have platform independent. From my perspective this looks like a relatively small change and it feels like missing a very good opportunity to fix it now instead of having to rewrite stuff in both map and connect later when this is not on top of the priority. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[app/components/Connect.js, line 152 at r2](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45#-L3hpiX7-fAiq0HgdLHd:-L3nBsp13wnXKr1X2muG:bz1ckly) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/4c6d971865662d6636c7ce7536fcbf4350696708/app/components/Connect.js#L152)):*
>No, i am saying that if the connect component wants to render something in the map component it should say that in something more logical than a magic offset of [0, 123], and something that can be made platform specific in a fairly easy way, 

This is possible to calculate but not until the component is mounted.  I'll take a look on Monday but keep in mind that I can't use ReactXP APIs yet in the Connect component so everything will be using DOM APIs at this time.

> the same goes for height/width

I agree, the initial implementation did exactly that. However, there were issues with this in `react-simple-maps` so I left the hardcoded dimensions for now. I can take another look but if this leads to the third-party library fixes this will postpone the whole release for a long time and we don't even have a backend for Android yet and everyone wants the maps :-)

> If zoomLevel is just depending on status, and if there is lat/lng i dont see why we should have some number that does not have a meaning if using another map. Enum or just levels that reflect what the purpose is (or just send the props needed for the logic to the map)

No I definitely not moving any logic relevant for Connect component into the Map but for now I'll switch the zoomLevel prop to string of three predefined zoom levels that Connect component can pass then.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

anderklander commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[app/components/Connect.js, line 152 at r2](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3hpiX7-fAiq0HgdLHd:-L3qZ2juQWWMhlRtuDSD:b-gmmqik) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/4c6d971865662d6636c7ce7536fcbf4350696708/app/components/Connect.js#L152)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

>No, i am saying that if the connect component wants to render something in the map component it should say that in something more logical than a magic offset of [0, 123], and something that can be made platform specific in a fairly easy way, 

This is possible to calculate but not until the component is mounted.  I'll take a look on Monday but keep in mind that I can't use ReactXP APIs yet in the Connect component so everything will be using DOM APIs at this time.

> the same goes for height/width

I agree, the initial implementation did exactly that. However, there were issues with this in `react-simple-maps` so I left the hardcoded dimensions for now. I can take another look but if this leads to the third-party library fixes this will postpone the whole release for a long time and we don't even have a backend for Android yet and everyone wants the maps :-)

> If zoomLevel is just depending on status, and if there is lat/lng i dont see why we should have some number that does not have a meaning if using another map. Enum or just levels that reflect what the purpose is (or just send the props needed for the logic to the map)

No I definitely not moving any logic relevant for Connect component into the Map but for now I'll switch the zoomLevel prop to string of three predefined zoom levels that Connect component can pass then.
</blockquote></details>

I think its much cleaner to just have the file extensions handle the platform dependencies than having to build logic inside the connect component to make special cases for different maps. Thats the main reason i want to move this from connect.

If its because of react-simple-maps we need to have hard coded size it sounds like another argument to put it inside map, since thats the place where the problem occurs? And when/if we have solved a proper size from connect to map we can remove the hard coded stuff?

If not all (size(at least the hard coded part, could still have the prop for future use), zoomlevel, offset) are moved to map or some other configuration file its not much of a difference to me since we will have to add special cases for each platform in connect anyway (or add some strange logic inside the new map to recalculate the values based on the react-simple-map implementation …)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[app/components/Connect.js, line 152 at r2](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3hpiX7-fAiq0HgdLHd:-L3sY8NrgtEGUNKJ4wMx:bre705g) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/4c6d971865662d6636c7ce7536fcbf4350696708/app/components/Connect.js#L152)):*
<details><summary><i>Previously, anderklander wrote…</i></summary><blockquote>

I think its much cleaner to just have the file extensions handle the platform dependencies than having to build logic inside the connect component to make special cases for different maps. Thats the main reason i want to move this from connect.

If its because of react-simple-maps we need to have hard coded size it sounds like another argument to put it inside map, since thats the place where the problem occurs? And when/if we have solved a proper size from connect to map we can remove the hard coded stuff?

If not all (size(at least the hard coded part, could still have the prop for future use), zoomlevel, offset) are moved to map or some other configuration file its not much of a difference to me since we will have to add special cases for each platform in connect anyway (or add some strange logic inside the new map to recalculate the values based on the react-simple-map implementation …)
</blockquote></details>

`react-simple-maps` is a bare bones component so we do a lot of other computations inside of the `Map` to make sure we get somewhat a decent performance that's why I am reluctant to adding anything else in there.

I think I have an idea.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

eriklarko commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-:-L40RgfWFd-KzYDOaTka:bnfp4nl)*
<!-- Sent from Reviewable.io -->

anderklander commented:
Yay!

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 21 of 29 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-:-L41D4WZFMoKKwniMGmH:b-rrhdxb)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 21 of 29 files reviewed at latest revision, 1 unresolved discussion.

---

*[app/components/Connect.js, line 152 at r2](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45#-L3hpiX7-fAiq0HgdLHd:-L41B-Wx5SK20f8gIZsE:b-3joyxi) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/4c6d971865662d6636c7ce7536fcbf4350696708/app/components/Connect.js#L152)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

`react-simple-maps` is a bare bones component so we do a lot of other computations inside of the `Map` to make sure we get somewhat a decent performance that's why I am reluctant to adding anything else in there.

I think I have an idea.
</blockquote></details>

I moved the specific SVG map implementation into `SvgMap` component and added a tiny `Map` component that tackles couple of quirks and provides the right zoom values etc..

Generally all your comments regarding zoom levels forced me too look in how I used to deal with this in the MapKit and basically found that using geographic span (think bbox) to display the desired area of the map is a better alternative to zoom levels since it's independent from the map dimensions and projection.

I left the "TODO" comments in code for reference. When that's done we'll switch over to geo center + geo span and drop external zoom control. 

This should provide an average solution for any kind of the Map as I see it and also help (in future) with the implementation of zooming on the country boundaries when connecting etc..

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 21 of 29 files reviewed at latest revision, 1 unresolved discussion.

---

*[app/components/Connect.js, line 152 at r2](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45#-L3hpiX7-fAiq0HgdLHd:-L41DrPR1OlxQKKckSe7:b-3kk4hi) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/4c6d971865662d6636c7ce7536fcbf4350696708/app/components/Connect.js#L152)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

I moved the specific SVG map implementation into `SvgMap` component and added a tiny `Map` component that tackles couple of quirks and provides the right zoom values etc..

Generally all your comments regarding zoom levels forced me too look in how I used to deal with this in the MapKit and basically found that using geographic span (think bbox) to display the desired area of the map is a better alternative to zoom levels since it's independent from the map dimensions and projection.

I left the "TODO" comments in code for reference. When that's done we'll switch over to geo center + geo span and drop external zoom control. 

This should provide an average solution for any kind of the Map as I see it and also help (in future) with the implementation of zooming on the country boundaries when connecting etc..


</blockquote></details>

Done. Please mark as satisfied and we'll merge if nobody minds. :)

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 21 of 29 files reviewed at latest revision, 1 unresolved discussion.

---

*[app/containers/ConnectPage.js, line 16 at r8](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45#-L41E0vkEOfvn3HYL44X:-L41E0vkEOfvn3HYL44Y:b74h7dk) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/b1694162650385b65e0e70e5fdc2681d0d91affd/app/containers/ConnectPage.js#L16)):*
> ```JavaScript
> import type { RelaySettingsRedux, RelayLocationRedux } from '../redux/settings/reducers';
> 
> function getRelayName(relaySettings: RelaySettingsRedux, relayLocations: Array<RelayLocationRedux>): string {
> ```

BTW: I moved the relay name extraction from the Component itself into the container. This way we don't need to provide the Connect with settings at all.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

eriklarko commented:
Reviewed 7 of 8 files at r8.
Review status: 28 of 29 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

eriklarko commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 28 of 29 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-:-L41HRSqOhIpeNWwKs5j:bnfp4nl)*
<!-- Sent from Reviewable.io -->

eriklarko commented:
Reviewed 1 of 8 files at r8.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

anderklander commented:
Review status: 28 of 29 files reviewed at latest revision, all discussions resolved.

---

*[app/components/Connect.js, line 152 at r2](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3hpiX7-fAiq0HgdLHd:-L41Oemu4Np7t4jczofx:b-v7gob2) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/4c6d971865662d6636c7ce7536fcbf4350696708/app/components/Connect.js#L152)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

Done. Please mark as satisfied and we'll merge if nobody minds. :)
</blockquote></details>

Well done!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 28 of 29 files reviewed at latest revision, all discussions resolved.

---

*[app/components/Connect.js, line 152 at r2](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45#-L3hpiX7-fAiq0HgdLHd:-L41TBG95MFl_PoYOkEu:b6x532s) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/4c6d971865662d6636c7ce7536fcbf4350696708/app/components/Connect.js#L152)):*
<details><summary><i>Previously, anderklander wrote…</i></summary><blockquote>

Well done!
</blockquote></details>

Thanks :+1: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/45)*
<!-- Sent from Reviewable.io -->
